src/Build/Logging/BinaryLogger/BuildEventArgsReader.cs

```diff
@@ -362,12 +362,14 @@ private BuildEventArgs ReadBuildFinishedEventArgs()
         {
             var fields = ReadBuildEventArgsFields();
             var succeeded = ReadBoolean();
+            var environmentProperties = ReadStringDictionary();
```

Comment:
need to check the fileFormatVersion and only ReadStringDictionary() if >= 15. Need to preserve current behavior when reading older binlogs.